packages/cozy-client/src/helpers/settings.js

```diff
@@ -0,0 +1,188 @@
+import CozyClient from '../CozyClient'
+import fetchPolicies from '../policies'
+import { Q } from '../queries/dsl'
+
+const defaultFetchPolicy = fetchPolicies.olderThan(10 * 1000)
+
+/**
+ * Query the cozy-app settings corresponding to the given slug and
+ * extract the value corresponding to the given `key`
+ *
+ * @param {CozyClient} client - Cozy client instance
+ * @param {string} slug - the cozy-app's slug containing the setting (can be 'instance' for global settings)
+ * @param {string} key - The name of the setting to retrieve
```

Comment:
Maybe having an array of keys could have been useful

Reply:
Why not, I'll try to add this

Reply:
I have this in my mind since yesterday but I'm struggling on choosing the best API.

### Declaration

For the declaration I imagine the following possibilities:

Use same hook for both single string param or multiple array param
```js
const { value } = useSetting('home', 'mysetting')
// value = 'some_value'

const { value } = useSetting('home', ['mysetting1', 'mysetting2'])
// value = {
//   mysetting1: 'some_value_1',
//   mysetting2: 'some_value_2'
// }
```

Use a different hooks for single and multiple params
```js
const { value } = useSetting('home', 'mysetting')
const { values } = useSettings('home', ['mysetting1', 'mysetting2'])
```

Use an object instead of an array to declare default values (but I we may [remove the default value thing](https://github.com/cozy/cozy-client/pull/1460#discussion_r1562642759), I think I can forget this option)
```js
const { value } = useSetting('home', 'mysetting')
const { values } = useSettings('home', {
  mysetting1: 'defaultValue1',
  mysetting2': 'defaultValue2'
})
```

### Mutation

Mutation is getting harder to approach

Should we return multiple save methods for each key?
```js
const { value, savemysetting1,  savemysetting2} = useSetting('home', ['mysetting1', 'mysetting2'])

savemysetting1('value1')
savemysetting2('value2')
```
or
```js
const { value, save } = useSetting('home', ['mysetting1', 'mysetting2'])

save.mysetting1('value1')
save.mysetting2('value2')
```
This constrains the API and so prevents errors, but sadly this also prevents mutating multiple settings at once

Should we return a save that takes an objet?
```js
const { value, save} = useSetting('home', ['mysetting1', 'mysetting2', 'mysetting3'])

savemysetting1({
  mysetting1: 'value1',
  mysetting3: 'value3'
})
```
This allow mutating multiple settings at once, but this makes the API less contraint and I'm wondering if we should allow inserting new keys or make a code that throw a specific error is the method is called with keys that are not present in the initial array?

Reply:
I made a proposal here: https://github.com/cozy/cozy-client/pull/1460/commits/dcb23c50e9c51f2db0cfd67ac1e22b736eb76078

(I still have to update documentation)

Usage would be:

### Hook

```js
const { value, save } = useSetting('home', ['mysetting1', 'mysetting2'])
...

// ✅ Valid
save({
  mySetting1: 'newValue1',
  mySetting2: 'newValue2'
})

// ✅ Valid
save({
  mySetting1: 'newValue1'
})

// ❌ Not valid because new key inserted
save({
  mySetting1: 'newValue1',
  mySetting2: 'newValue2',
  mySetting3: 'newValue3'
})

```

### Get
```js
const result = await getSetting(client, 'home', ['some_key'])
```


### Save
```js
// With raw value
await client.saveAfterFetchSetting(
  'instance',
  {
    some_instance_key: 'some_new_instance_value',
    some_instance_key2: 'some_new_instance_value_2'
  }
)


// With a method
await client.saveAfterFetchSetting(
  'instance',
  currentValue => ({
    ...currentValue,
    some_instance_key: currentValue.some_instance_key + 1,
    some_instance_key2: 'some_new_instance_value_2'
  }),
  ['some_instance_key']
)
```

When using a method we must pass a `setterKeys` parameters so the save method can still extract correct values. I would prefer not to have this parameter but I did not find a better way 😕

Reply:
Note that I typed those methods to prevent mistakes when using them. Here is how the linter and the auto-complete features behave:
![image](https://github.com/cozy/cozy-client/assets/1884255/8f17023a-fe4e-418b-b80b-0d91d6e43cb4)

Reply:
I updated the cozy-home PR with the new `multiple-settings` API: https://github.com/cozy/cozy-home/pull/2117/commits/e20239d6dee132ba566ca3f0a3e6ca075acab56c